src/platform/Darwin/DnssdImpl.cpp

```diff
@@ -219,25 +293,48 @@ static void OnGetAddrInfo(DNSServiceRef sdRef, DNSServiceFlags flags, uint32_t i
     ReturnOnFailure(MdnsContexts::GetInstance().Has(sdCtx));
     LogOnFailure(__func__, err);
 
+    sdCtx->domainName = GetDomainFromHostName(hostname);
+    if (sdCtx->domainName.empty())
+    {
+        ChipLogError(Discovery, "Mdns: Domain name is not set");
+        return;
+    }
     if (kDNSServiceErr_NoError == err)
     {
-        sdCtx->OnNewAddress(interfaceId, address);
+        std::pair<uint32_t, std::string> key = std::make_pair(interfaceId, sdCtx->domainName);
+        sdCtx->OnNewAddress(key, address);
     }
 
     if (!(flags & kDNSServiceFlagsMoreComing))
     {
         VerifyOrReturn(sdCtx->HasAddress(), sdCtx->Finalize(kDNSServiceErr_BadState));
-        sdCtx->Finalize();
+
+        if (sdCtx->domainName.compare(kOpenThreadDot) == 0)
+        {
+            ChipLogProgress(Discovery, "Mdns: Resolve completed on the open thread domain.");
+            sdCtx->Finalize();
+        }
+        else if (sdCtx->domainName.compare(kLocalDot) == 0)
+        {
+            ChipLogProgress(
+                Discovery,
+                "Mdns: Resolve completed on the local domain. Starting a timer for the open thread resolve to come back");
+            if (!hasOpenThreadTimerStarted)
```

Comment:
Why do we treat the local and openthread domains differently here, in the sense that we wait longer for the open thread resolve if necessary, but not the other way around if that one were to come back first? If that's the correct logic a comment explaining why we do that would be good.

Reply:
Yeah i will add a comment. The reason for this is resolution on open thread take a little longer and in our discussions with Ted we decided to give it 250 millisecs more before we wrap up. I will add a meaningful comment

Reply:
done.